src/main/scala/za/co/absa/commons/reflect/ReflectionUtils.scala

```diff
@@ -81,16 +82,47 @@ object ReflectionUtils {
   }
 
   /**
-    * Returns an object instance with the specified name.
-    * Similar to as Class.forName() returns a Class instance by name, this method returns a Scala object instance by name.
-    *
-    * @param name fully qualified object instance name
+   * Returns an object instance with the specified name.
+   * Equivalent of Class.forName() for objects, as Class.forName() returns a Class instance by name,
+   * this method returns a Scala object instance by name.
+   *
+   * @param name fully qualified object instance name
+   * @return a singleton instance of a class with given name
     */
   def objectForName[T <: AnyRef](name: String): T = {
     val moduleSymbol = mirror.staticModule(name)
     mirror.reflectModule(moduleSymbol).instance.asInstanceOf[T]
   }
 
+  /**
+   * Returns an object instance with the specified name.
+   * Equivalent of Class.forName() for objects, as Class.forName() returns a Class instance by name,
+   * this method returns a Scala object instance by name.
+   * Provides a human-understandable description of what went wrong in case of exception.
+   *
+   * @param name fully qualified object instance name
+   * @return a singleton instance of a class with given name
+   */
+  @throws[IllegalArgumentException]
+  def objectForNameWithDescriptiveException[T <: AnyRef : ClassTag : TypeTag](name: String): T = {
+    Try(mirror.staticModule(name)) match {
+      case Success(module) =>
+        val reflectiveMirror = mirror.reflectModule(module)
+        Try(reflectiveMirror.instance) match {
+          case Success(instance) =>
+            instance match {
+              case singleton: T => singleton
+              case _ =>
+                throw new IllegalArgumentException(s"Class '$name' is not an instance of '${typeOf[T]}'")
```

Comment:
Why is `IllegalArgumentException` more descriptive than a reflection exception thrown from a reflection operation?

Reply:
The descriptiveness in the name of this function comes mostly from the case when provided `name` is not a singleton. In that case, `objectForName` throws `java.lang.ClassNotFoundException: <name>`, and `objectForNameWithDescriptiveException` throws `IllegalArgumentException: Class '<name>' is not a singleton`.

However, the more I think about it, the more I'm convinced that this function is not needed, and I'll probably remove it from this PR.